service/controller/resource/instance/create_cordon_old_workers.go

```diff
@@ -165,3 +165,59 @@ func sortNodesByTenantVMState(nodes []corev1.Node, instances []compute.VirtualMa
 
 	return
 }
+
+func (r *Resource) getK8sWorkerNodeForInstance(ctx context.Context, customObject providerv1alpha1.AzureConfig, instance compute.VirtualMachineScaleSetVM) (*corev1.Node, error) {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	name := key.WorkerInstanceName(customObject, *instance.InstanceID)
+
+	nodeList, err := cc.Client.TenantCluster.K8s.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	nodes := nodeList.Items
+
+	for _, n := range nodes {
+		if n.GetName() == name {
+			return &n, nil
+		}
+	}
+
+	// Node related to this instance was not found.
+	return nil, nil
+}
+
+func (r *Resource) isWorkerInstanceFromPreviousRelease(ctx context.Context, customObject providerv1alpha1.AzureConfig, instance compute.VirtualMachineScaleSetVM) (*bool, error) {
+	t := true
+	f := false
```

Comment:
What's with the boolean pointers? :smile:  Just returning true or false wasn't enough?